# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoStreamFactory.h
# Commit: 7d270e96dfbf
# Full Hash: 7d270e96dfbfcc5992f9a1704e7d2229d1a645f3
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 4
# Description:
#   Bug 1654112 - Adapt VideoConduit to the new libwebrtc version. r=ng,bwc
#   
#   This hooks up VideoReceiveStream::Config with a decoder factory and
#   VideoSendStream::Config with both a bitrate allocator factory and an encoder
#   factory.
# ==============================================================================

diff -r 2f1e8f5b2fc2 -r 7d270e96dfbf dom/media/webrtc/libwebrtcglue/VideoStreamFactory.h
--- a/dom/media/webrtc/libwebrtcglue/VideoStreamFactory.h	Mon Feb 08 17:23:44 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/VideoStreamFactory.h	Mon Feb 08 18:15:07 2021 +0100
@@ -27,6 +27,10 @@
     int max_bitrate_bps;
   };
 
+  static ResolutionAndBitrateLimits GetLimitsFor(unsigned int aWidth,
+                                                 unsigned int aHeight,
+                                                 int aCapBps = 0);
+
   VideoStreamFactory(VideoCodecConfig aConfig,
                      webrtc::VideoCodecMode aCodecMode, int aMinBitrate,
                      int aStartBitrate, int aPrefMaxBitrate,